src/core/InstancedMesh2.ts

```diff
@@ -269,14 +269,18 @@ export class InstancedMesh2<
       this.performFrustumCulling(shadowCamera, camera);
     }
 
-    this.matricesTexture.update(renderer);
-    this.colorsTexture?.update(renderer);
-    this.uniformsTexture?.update(renderer);
-    this.boneTexture?.update(renderer);
-    // TODO convert also morph texture to squared texture to use partial update
+    if (renderer instanceof WebGLRenderer) {
```

Comment:
Instead of these `if` `else` blocks, I think it should be possible to use the nodes system to write TSL that can compile to work with either renderer